papermill_origami/noteable_dagstermill/assets.py

```diff
@@ -0,0 +1,314 @@
+    metadata: Optional[Mapping[str, Any]] = None,
+    config_schema: Optional[Union[Any, Dict[str, Any]]] = None,
+    required_resource_keys: Optional[Set[str]] = None,
+    resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
+    description: Optional[str] = None,
+    partitions_def: Optional[PartitionsDefinition] = None,
+    op_tags: Optional[Dict[str, Any]] = None,
+    group_name: Optional[str] = None,
+):
+    """Creates a Dagster asset for a Noteable notebook.
+
+    Arguments:
+        name (str): The name for the asset
+        notebook_id (str): The id of the Noteable notebook
+        key_prefix (Optional[Union[str, Sequence[str]]]): If provided, the asset's key is the
+            concatenation of the key_prefix and the asset's name, which defaults to the name of
+            the decorated function. Each item in key_prefix must be a valid name in dagster (ie only
+            contains letters, numbers, and _) and may not contain python reserved keywords.
+        ins (Optional[Mapping[str, AssetIn]]): A dictionary that maps input names to information
+            about the input.
+        non_argument_deps (Optional[Union[Set[AssetKey], Set[str]]]): Set of asset keys that are
+            upstream dependencies, but do not pass an input to the asset.
+        config_schema (Optional[ConfigSchema): The configuration schema for the asset's underlying
+            op. If set, Dagster will check that config provided for the op matches this schema and fail
+            if it does not. If not set, Dagster will accept any config provided for the op.
+        metadata (Optional[Dict[str, Any]]): A dict of metadata entries for the asset.
+        required_resource_keys (Optional[Set[str]]): Set of resource handles required by the notebook.
+        description (Optional[str]): Description of the asset to display in Dagit.
+        partitions_def (Optional[PartitionsDefinition]): Defines the set of partition keys that
+            compose the asset.
+        op_tags (Optional[Dict[str, Any]]): A dictionary of tags for the op that computes the asset.
+            Frameworks may expect and require certain metadata to be attached to a op. Values that
+            are not strings will be json encoded and must meet the criteria that
+            `json.loads(json.dumps(value)) == value`.
+        group_name (Optional[str]): A string name used to organize multiple assets into groups. If not provided,
+            the name "default" is used.
+        resource_defs (Optional[Mapping[str, ResourceDefinition]]):
+            (Experimental) A mapping of resource keys to resource definitions. These resources
+            will be initialized during execution, and can be accessed from the
+            context within the notebook.
+    """
+
+    check.str_param(name, "name")
+    check.str_param(notebook_id, "notebook_id")
+
+    # TODO - make these parameterizable/from env vars?
```

Comment:
I'm not 100% sure if it makes to hardcode these values (noteable:// and the url prefix) - I can instead pull them from environment variables with these values as the defaults if you might support custom domains in the future or have other plans for the path prefix and url prefix

Reply:
You can check `NOTEABLE_DOMAIN`, falling back to `""https://app.noteable.io"` as a default: https://docs.noteable.io/product-docs/work-with-notebooks/scheduling-and-orchestration/build-your-own.

The notebook_id needs to be passed in imo